plan/stats.go

```diff
@@ -118,57 +123,82 @@ func (ds *DataSource) getStatsByFilter(conds expression.CNFExprs) *statsInfo {
 	return profile.scale(selectivity)
 }
 
-func (ds *DataSource) deriveStats() *statsInfo {
+func (ds *DataSource) deriveStats() (*statsInfo, error) {
 	// PushDownNot here can convert query 'not (a != 1)' to 'a = 1'.
 	for i, expr := range ds.pushedDownConds {
 		ds.pushedDownConds[i] = expression.PushDownNot(nil, expr, false)
 	}
 	ds.statsAfterSelect = ds.getStatsByFilter(ds.pushedDownConds)
-	return ds.statsAfterSelect
+	for _, path := range ds.possibleAccessPaths {
```

Comment:
how about:
```go
if !ds. possibleAccessPaths[0].isRowID {
    panic("table path should always be the first access path"
}
err := ds.deriveTablePathStats(path)
...
for _, path := range ds.possibleAccessPaths[1:] {
    err := ds.deriveIndexPathStats(path)
    ...
}
```

Reply:
No, the first element can be index path.